### PR TITLE
Define new function islandora_sort_ordered

### PR DESCRIPTION
**Pull request by jonathangreen**
**JIRA Ticket**:
https://jira.duraspace.org/browse/ISLANDORA-1887

* Other Relevant Links (Google Groups discussion, related pull requests, Release pull requests, etc.)
https://www.drupal.org/node/2756297

# What does this Pull Request do?

This function can replace calls to uasort with drupal_sort_weight. This
function will return a consistant sort order between PHP5 and PHP7 for
elements of the same weight by preserving the order of elements with the
same weight.

# What's new?

Under both PHP5 and PHP7 the order of elements with the same weight are
not guarenteed, and in practice the order of elements returned is
different between PHP5 and PHP7. This could cause some issues where the
order is changed depending on PHP version. This function gives us the
option of a consistant sort order.

This pull request also defines some new unit tests for the new function.

# How should this be tested?

This pull request updates ingest steps and derivative creation to use the new function.

# Additional Notes:

The sort order of islandora_sort_ordered is likely to be different from both the sort order of uasort in PHP5 and uasort in PHP7 because the sort order of equal keys is undefined.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

File: tests/utilities.test [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ */
+class IslandoraSortOrderedTestCase extends DrupalUnitTestCase {
+
+  /**
+   * Return info about the tests.
+   */
+  public static function getInfo() {
+    return array(
+      'name' => 'islandora_sort_ordered()',
+      'description' => 'Performs unit tests on islandora_sort_oredered().',
```
Comment (jordandukart) [inline tests/utilities.test]: Spelling?

**Comment by jordandukart** [comment]
If/when gets approved to core will the calls to these functions point to that instead?

**Comment by jonathangreen** [comment]
On second thought I re-rolled my patch to core drupal so the function is identical to this pull request. That way if it ever does make it into Drupal core then we can just replace `islandora_sort_ordered` with `drupal_sort_ordered`.
https://www.drupal.org/node/2834022

**Comment by jonathangreen** [comment]
Fixed spelling.

**Comment by jordandukart** [comment]
Looking at this today.

**Comment by jordandukart** [comment]
Chatted with @jonathangreen about implications and whether we need a stable sort. 

Summary:
-PHP's `uasort` is defined as not being stable and runs into issues as described where the comparator returns a 0 value.
-The core cases that are not form related (Solr metadata/Ingest Steps/Derivatives/Paged content) etc should have explicit weights set when being passed out to be sorted.

Will bring this up at the next committers to get a consensus on direction.

**Comment by jonathangreen** [comment]
Closing issue: discussed in committers, we don't care if order changes between PHP5 and PHP7. In practice it hasn't caused any issues.

**Pull request closed without merging**